src/App.jsx

```diff
@@ -14,33 +16,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_LENGTH = 'length';
+const SORT_FIELD_ALPHABETICALLY = 'alphabetically';
+const SORT_FIELD_REVERSE = 'reverse';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sortField, setSortField] = useState('');
+  const [sortFieldReverse, setSortFieldReverse] = useState('');
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const reset = () => {
+    setVisibleGoods(goodsFromServer);
+    setSortField('');
+    setSortFieldReverse('');
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const sortByLength = () => {
+    if (sortFieldReverse === SORT_FIELD_REVERSE) {
+      setVisibleGoods(
+        [...visibleGoods].sort((good1, good2) => good2.length - good1.length),
+      );
+    } else {
+      setVisibleGoods(
+        [...visibleGoods].sort((good1, good2) => good1.length - good2.length),
+      );
+    }
+
+    setSortField(SORT_FIELD_LENGTH);
+  };
+
+  const reverseSort = () => {
+    setVisibleGoods([...visibleGoods].reverse());
+    if (sortFieldReverse === '') {
+      setSortFieldReverse(SORT_FIELD_REVERSE);
+    } else {
+      setSortFieldReverse('');
+    }
+  };
+
+  const sortByAlphabet = () => {
+    if (sortFieldReverse === SORT_FIELD_REVERSE) {
+      setVisibleGoods(
+        [...visibleGoods].sort((good1, good2) => good2.localeCompare(good1)),
+      );
+    } else {
+      setVisibleGoods(
```

Comment:
These 3 sorting functions need to be combined into one, as they are very similar, and sorting is better done in a switch case construction

Reply:
```
function getPreparedGoods(goods, { sortField, reverseField }) {
  const preparedGoods = [...goods];

  if (sortField) {
    preparedGoods.sort((good1, good2) => {
      switch (sortField) {
        .........
      }
    }
  }
}
```